benchmarks/babelstream/CMakeLists.txt

```diff
@@ -19,9 +19,36 @@ if(NOT TARGET alpaka::alpaka)
     endif()
 endif()
 
-alpaka_add_executable(${PROJECT_NAME} src/main.cpp src/Stream.h src/AlpakaStream.cpp src/AlpakaStream.h)
-target_compile_definitions(${PROJECT_NAME} PUBLIC ALPAKA)
-target_link_libraries(${PROJECT_NAME} PUBLIC alpaka::alpaka)
-set_target_properties(${PROJECT_NAME} PROPERTIES FOLDER benchmarks)
 
-# add_test(NAME ${_TARGET_NAME} COMMAND ${_TARGET_NAME})
+set(_TARGET_NAME "babelstream")
+append_recursive_files_add_to_src_group("src/" "src/" "cpp" _FILES_SOURCE)
+
+alpaka_add_executable(
+    ${_TARGET_NAME}
+    ${_FILES_SOURCE})
+
+target_include_directories(
+ ${_TARGET_NAME}
+ PRIVATE "src")
+
+target_link_libraries(
+    ${_TARGET_NAME}
+    PRIVATE common)
+
+#Run as a ctest
+if(alpaka_CI)
+    # Only run for release builds since this is a benchmark
+    if(CMAKE_BUILD_TYPE STREQUAL "Release")
+        if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux" OR ${CMAKE_SYSTEM_NAME} STREQUAL "Darwin")
+           # To prevent CI test timeout, the number of samples needs to be limited. In the catch2 benchmarking; the code is run samples x iterations times.
+           add_test(NAME ${_TARGET_NAME} COMMAND sh -c "${CMAKE_CURRENT_BINARY_DIR}/${_TARGET_NAME} ${_alpaka_TEST_OPTIONS}  --benchmark-samples 1 -r xml | tee BabelStreamBenchmarkResults.xml")
+        elseif(${CMAKE_SYSTEM_NAME} STREQUAL "Windows")  
+           # To prevent CI test timeout, the number of samples needs to be limited. In the catch2 benchmarking; the code is run samples x iterations times.
+           set_target_properties(${_TARGET_NAME} PROPERTIES FOLDER benchmarks/babelstream)
+           add_test(NAME ${_TARGET_NAME} COMMAND ${_TARGET_NAME} --benchmark-samples 1)
```

Comment:
Use this also for Linux and Darwin

Reply:
why ? Don't we need xml?

Reply:
ok, done. ( XML output file is not generated currently, because CI is not a proper environment for running the benchmark to measure performance.  When the benchmark is run exclusively, XML file could be generated by changing cmake or while running the benchmark from command line)